homeassistant/components/switchbot/sensor.py

```diff
@@ -52,6 +66,9 @@ async def async_setup_entry(
         DATA_COORDINATOR
     ]
 
+    if not coordinator.data.get(entry.unique_id):
+        raise PlatformNotReady
+
```

Comment:
This should not happen.

That's what this is for:
https://github.com/home-assistant/core/blob/512bdac7245e85f2faabe20800e1ae0628faa808/homeassistant/components/switchbot/__init__.py#L83

Reply:
hm, it happened with my sensor after https://github.com/home-assistant/core/pull/59584 (I have one that's inside the fridge, so although it actually works, it's very unreliable and works like 20% of the times).
I'm not very familiar with the codebase to understand why this would happen after my change though.

Reply:
Hi @lestrozi,

This might actually be a good idea. The devices might be out of range and not respond to advertisements. (intermittently if the signal is bad).

This would result in the first refresh working, but might not have data for the spesific switchbot. I'm working on some or other way to assume a state until the data is received. (Kind of how the shelly integration works with sleeping devices)